### ISSUE TITLE
Module resolution error when using components on npm packages

### ISSUE THREAD

**Issue by armandoisaac** [muenzpraeger/create-lwc-app#590]
### Summary

We have a project that was working before the changes on the module resolution. After the changes were merged and deployed, our project stopped working. We figured out that we need a new file called `lwc.config.js` on the root folder of the project where we need to define which components will be exported and the local modules folder. We figured out that we need to follow the RFC located at https://rfcs.lwc.dev/rfcs/lwc/0020-module-resolution. 

After all this changes, we still can't reference our npm package project into our "client" project as it throws errors regarding missing components. Our folder structure looks like this:

NPM package structure (we called this package `@test/lightning`):
```
src
  |--> modules
             |--> lightning
                           |--> button
                                     |--> button.js
                                     |--> button.html
                           |--> primitiveButton
                                    |--> primitiveButton.js
                                    |--> primitiveButton.html
                           |--> primitiveIcon
                                    |--> primitiveIcon.js
                                    |--> primitiveIcon.html
                           |--> utils
                                    |--> utils.js
                           |--> utilsPrivate
                                    |--> utilsPrivate.js
```

NPM package `lwc.config.js` looks like this:
```
{
    "expose": [
        "lightning/button",
        "lightning/primitiveButton",
        "lightning/primitiveIcon",
        "lightning/utils",
        "lightning/utilsPrivate"
    ],
    "modules": [
        {
            "dir": "src/modules"
        }
    ]
}
```

Running this npm package project by itself works without any issue and we can use the tag `<lightning-button />`.

On our "client" project, we have installed the `@test/lightning` npm package and our folder structure looks like the boilerplate used by the `create-lwc-app`:
```
src
  |--> modules
             |--> my
                           |--> app
                                     |--> app.js
                                     |--> app.html
```

The `lwc.config.js` file looks like this: 

```
{
    "modules": [
        {
            "npm": "@test/lightning"
        }
    ]
}
```

The `app.html` file looks like:

```
<template>
    <div class="center greeting">
        <lightning-button variant="base" label="Base" title="Looks like a link" class="slds-m-left_x-small">
        </lightning-button>
    </div>
</template>
```

After running using `npm run watch`, an error is thrown and the app won't run:

```
ERROR in ./node_modules/@test/lightning/src/modules/lightning/button/button.js
Module not found: Error: Can't resolve 'lightning/primitiveButton' in '/Users/armandoruiz/Projects/new-lwc/resources/node_modules/@test/lightning/src/modules/lightning/button'
 @ ./node_modules/@test/lightning/src/modules/lightning/button/button.js 34:0-57 37:22-38
 @ ./src/modules/my/app/app.html
 @ ./src/modules/my/app/app.js
 @ ./src/index.js
 @ multi ./node_modules/error-overlay-webpack-plugin/lib/entry-basic.js ./node_modules/error-overlay-webpack-plugin/lib/entry-devserver.js? ./src/index.js
```

After looking into the `module-resolver`, we have found that indeed it locates the `lightning/button`, but not the components that this component references (lightning/primitiveButton and lightning/utils):

Module-resolver output when trying to locate the `lightning/button`:
```
{
  entry: '/Users/armandoruiz/Projects/new-lwc/resources/node_modules/@test/lightning/src/modules/lightning/button/button.js',
  specifier: 'lightning/button',
  scope: '/Users/armandoruiz/Projects/new-lwc/resources/node_modules/@test/lightning'
}
```

We also tried to add this to the `lwc-services.config.js` file:

```
    localModulesDirs: [
        './node_modules/@test/lightning/src/modules'
    ]
```

### Steps To Reproduce:

1.  Create a new lwc project using `npm create-lwc-app`.
2. Add 2 components. One component should reference the other component (parent/child)
3. Configure the `lwc.config.js` file and expose both parent and child component.
4. Create a second new lwc project.
5. Install the first project into this new project as an npm module.
6. Configure the `lwc.config.js` of the second project to use the exposed components from the first project
7. Use the parent component from the first project in the second project.

### Expected result

The app should work and we should be able to use the components that are located in the npm package and it should also resolve child components.

### Actual result

Resolver throws errors notifying that it can't locate other components referenced in the npm package.

### Additional information

NPM package running on standalone mode:
<img width="793" alt="npm-example" src="https://user-images.githubusercontent.com/6314810/83074463-11bd2a80-a027-11ea-8c61-0ad781ed586f.png">

Errors on client project
<img width="1236" alt="errors" src="https://user-images.githubusercontent.com/6314810/83074612-4e892180-a027-11ea-91e2-f04f17ab37c0.png">

Also, it would be nice if all this changes and configurations can be documented, we spent some days trying to figure out what changed and where this `lwc.config.js` file came from.

**Comment by muenzpraeger** [comment]
Thanks for opening an issue.

First, try the beta channel on npmjs.com, the last version of `lwc-services` from two weeks ago should unblock you.

The implementation of the module resolution is documented for v2.0.7 [in the changelog](https://github.com/muenzpraeger/create-lwc-app/blob/master/CHANGELOG.md#2020-04-27). That consuming 3rd party npm packages worked before was never supported by `lwc-services`, at it used an internal mechanism of the LWC `module-resolver`.

For documentation of LWC core features, like `lwc.config.json`, I'm referring to the LWC team and lwc.dev, as all functionality related to LWC in the tooling here depends on what is supported there.

**Comment by Gr8Gatsby** [comment]
I verified that using the 2.0.8-beta.1 works.

**Comment by kostadin-stoev** [comment]
Had the same issue. Can confirm as well that 2.0.8-beta.1 works.

**Comment by armandoisaac** [comment]
Thanks @muenzpraeger , the beta works perfect. Do you have an estimate on when it is going to be promoted?

**Comment by muenzpraeger** [comment]
@armandoisaac I plan/hope by end of this week latest.

**Comment by armandoisaac** [comment]
I can confirm the issue has been fixed with release 2.1.0.

Thanks for your help René
